components/rate/demo/character-function.md

```diff
@@ -13,23 +13,26 @@ title:
 
 Can customize each character using `(RateProps) => ReactNode`.
 
-```jsx
+```tsx
+import React from 'react';
 import { Rate } from 'antd';
 import { FrownOutlined, MehOutlined, SmileOutlined } from '@ant-design/icons';
 
-const customIcons = {
+const customIcons: Record<number, React.ReactNode> = {
   1: <FrownOutlined />,
   2: <FrownOutlined />,
   3: <MehOutlined />,
   4: <SmileOutlined />,
   5: <SmileOutlined />,
 };
 
-export default () => (
+const App: React.FC = () => (
   <>
-    <Rate defaultValue={2} character={({ index }) => index + 1} />
+    <Rate defaultValue={2} character={({ index }: { index: number }) => index + 1} />
```

Comment:
https://github.com/ant-design/ant-design/blob/c92e50833363d5cfd28a52923e9db827178292bc/components/rate/index.tsx#L19

这个定义要更新。

Reply:
This modification is causing the issue: 
````
Type '({ index }: { index: number; }) => ReactNode' is not assignable to type 'ReactNode | ((props: StarProps) => ReactNode)'.
  Type '({ index }: { index: number; }) => ReactNode' is not assignable to type '(props: StarProps) => ReactNode'.
    Types of parameters '__0' and 'props' are incompatible.
      Type 'StarProps' is not assignable to type '{ index: number; }'.
        Types of property 'index' are incompatible.
          Type 'number | undefined' is not assignable to type 'number'.
            Type 'undefined' is not assignable to type 'number'.ts(2322)
````
Thus making the [whole demo](https://github.com/ant-design/ant-design/blob/6a0e8a6c9c6c290f80677444ae136283ec5da414/components/rate/demo/character-function.tsx#L15C15-L15C15) unusable (see [example](https://codesandbox.io/p/sandbox/customize-character-antd-5-12-3-336334?file=%2Fdemo.tsx%3A17%2C11))
This could be fixed like this:

````typescript
<Rate defaultValue={2} character={({ index }: { index?: number }) => index !== undefined && index + 1} />
````